src/commands/zunda.rs

```diff
@@ -73,21 +108,113 @@ async fn leave(ctx: &Context, msg: &Message) -> CommandResult {
-                .await,
-        );
+        if let Err(e) = remove_all_channels(ctx, guild_id).await {
+            check_msg(
+                msg.channel_id
+                    .say(&ctx.http, format!("Failed: {:?}", e))
+                    .await,
+            );
+        } else {
+            check_msg(
+                msg.channel_id
+                    .say(
+                        &ctx.http,
+                        "サヨナラなのだ!また必要になったら`vc`で呼ぶのだ!",
+                    )
+                    .await,
+            );
+        }
     } else {
         check_msg(msg.reply(ctx, "Not in a voice channel").await);
     }
 
     Ok(())
 }
 
+#[command]
+#[aliases("add")]
+#[description = "読み上げ対象のチャンネルを追加するのだ!"]
+#[only_in(guilds)]
+async fn listen(ctx: &Context, msg: &Message) -> CommandResult {
+    let guild = msg.guild(&ctx.cache).unwrap();
+    let guild_id = guild.id;
+
+    if has_handler(ctx, guild_id).await {
+        if let Err(e) = add_channels(ctx, guild_id, vec![msg.channel_id]).await {
+            check_msg(
+                msg.channel_id
+                    .say(&ctx.http, format!("Failed: {:?}", e))
+                    .await,
+            );
+        } else {
+            check_msg(msg.reply(ctx, "VCの読み上げ対象に追加したのだ!").await);
+        }
+    }
+
+    Ok(())
+}
+
+#[command]
+#[description = "読み上げ対象のチャンネルを確認するのだ!"]
+#[only_in(guilds)]
+async fn list(ctx: &Context, msg: &Message) -> CommandResult {
+    let app_state = app_state::get(ctx).await.unwrap();
+    let subscribe_channels = app_state.read().await.subscribe_channels.clone();
+
+    let guild = msg.guild(&ctx.cache).unwrap();
+    let guild_id = guild.id;
+
+    let channels = subscribe_channels.get(&guild_id).unwrap_or(&vec![]).clone();
+
+    check_msg(
+        msg.reply(
+            ctx,
+            format!(
+                "読み上げ対象は\n {} \nなのだ!",
+                channels
+                    .iter()
+                    .map(|c| format!(" <#{}> ", c))
+                    .collect::<Vec<String>>()
+                    .join("\n")
+            ),
+        )
+        .await,
+    );
+
+    Ok(())
+}
+
+#[command]
+#[aliases("remove")]
+#[description = "読み上げ対象のチャンネルを削除するのだ!"]
+#[only_in(guilds)]
+async fn listen_remove(ctx: &Context, msg: &Message) -> CommandResult {
+    let guild = msg.guild(&ctx.cache).unwrap();
+    let guild_id = guild.id;
+
+    if has_handler(ctx, guild_id).await {
+        if let Err(e) = remove_channel(ctx, guild_id, msg.channel_id).await {
+            check_msg(
+                msg.channel_id
+                    .say(&ctx.http, format!("Failed: {:?}", e))
+                    .await,
+            );
+        } else {
+            check_msg(msg.reply(ctx, "VCの読み上げ対象から削除したのだ!").await);
+        }
+    }
+
+    Ok(())
+}
+
+async fn has_handler(ctx: &Context, guild_id: GuildId) -> bool {
+    let manager = songbird::get(ctx)
+        .await
+        .expect("Songbird Voice client placed in at initialisation.")
+        .clone();
```

Comment:
```suggestion
        .expect("Songbird Voice client placed in at initialisation.");
```
`songbird::get` は songbird が保持している `Arc` を `clone` して返してきています. 不要な `clone` かと